wikidatarefisland/pipes/item_mapping_matcher_pipe.py

```diff
@@ -0,0 +1,48 @@
+from wikidatarefisland.pipes import AbstractPipe
+
+
+class ItemMappingMatcherPipe(AbstractPipe):
+    """A pipe segment to match potential references with statement data."""
+
+    def __init__(self, mapping, whitelisted_external_identifiers):
+        """Instantiate the pipe
+
+        Arguments:
+            matchers {wikidatarefisland.data_model.ValueMatchers} --
+                A static class with value matcher functions
+        """
+        self.mapping = mapping
+        self.whitelisted_external_identifiers = whitelisted_external_identifiers
+
+    def flow(self, potential_match):
+        """Applies transformations to data flow
+
+        Arguments:
+            potential_match {dict} -- A potential statement - reference match to examine.
+                See: See: https://github.com/wmde/reference-island#statement-reference-blob
+
+        Returns:
+            The input, this class just observes.
+        """
+        if potential_match['statement']['datatype'] != 'wikibase-item':
+            return []
+        item_id = potential_match['statement']['value']['numeric-id']
+        ext_id_property = None
+        for pid in potential_match['reference']['referenceMetadata']:
+            if pid not in self.whitelisted_external_identifiers:
+                continue
+            ext_id_property = pid
+            break
+        if not ext_id_property:
+            return []
```

Comment:
Same comment as the one on `item_statistical_matcher_pipe.py`

Reply:
I responded there, I definitely thing we should pick it up but I think it's outside of scope of this PR, we should track it and then pick it up for the next sprint. What do you think?